### ISSUE TITLE
[Exception] [__NSURLBackgroundSession getAllTasksWithCompletionHandler:]: unrecognized selector sent to instance 0x183022f0

### ISSUE THREAD

**Issue by AeonDave** [SimonSimCity/Xamarin-CrossDownloadManager#28]
I got an exception while creating File

String.downloadurl is a string like http://www.foo.com/file

i'm loading it on an iPad mini with last ios os
could you help me?

```
public Downloader()
		{
			File = CrossDownloadManager.Current.CreateDownloadFile(
				Strings.downloadUrl
			);
		}
.......................
              Downloader d = new Downloader();
```
[Exception]

 -[__NSURLBackgroundSession getAllTasksWithCompletionHandler:]: unrecognized selector sent to instance 0x183022f0
 Xamarin.iOS: Received unhandled ObjectiveC exception: NSInvalidArgumentException -[__NSURLBackgroundSession getAllTasksWithCompletionHandler:]: unrecognized selector sent to instance 0x183022f0

**Comment by AeonDave** [comment]
this is the crash poit inside the plugin

```
_session.GetAllTasks((NSUrlSessionTask[] tasks) => {
				foreach (var task in tasks) {
					if (task is NSUrlSessionDownloadTask) {
						Queue.Add(new DownloadFileImplementation((NSUrlSessionDownloadTask)task));
					}
				}
			});
```

**Comment by SimonSimCity** [comment]
Thanks for reporting the error. Do you also have a full stacktrace?

If you've time, it would also help a lot if you create a fork of this project and manipulate the sample project so it crashes - but with the fewest changes :wink:

**Comment by AeonDave** [comment]
really sorry, is a native crash, i cannot print the error. this is all i get from console.
this crash fire only with an ipad mini. that's strange anyway.

**Comment by SimonSimCity** [comment]
@AeonDave What version of iOS is installed on the iPad mini?

**Comment by AeonDave** [comment]
8.1.2 (12B440)

**Comment by SimonSimCity** [comment]
Right ... haven't thought about that when writing this code. `_session.GetAllTasks()` is only available at iOS9+, but I've already found another method that is available since iOS7+.

**Comment by AeonDave** [comment]
Great!
is there any possibility to POST data  to gain a file? (not through header)
es:
i have a webservice that deliver a file only if you post a token and an id

**Comment by SimonSimCity** [comment]
Please create a new issue for this request.

Can you confirm that it works now if you build the sample project?

**Comment by AeonDave** [comment]
i can confirm, it works like a charm.

i'll open another issue for that ok?

2016-08-31 8:35 GMT+02:00 Simon Schick <notifications@github.com>:

**Comment by SimonSimCity** [comment]
Fixed in version 0.7.1 - now available on https://www.nuget.org/packages/Xam.Plugins.DownloadManager/
